draft-ietf-taps-interface.md

```diff
@@ -737,23 +737,45 @@ early binding when required, for example with some Network Address Translator
 
 ### Using Multicast Endpoints
 
-Specifying a multicast group address on a Local Endpoint will indicate to the Transport
-Services system that the resulting connection will be used to receive multicast messages. The
-Remote Endpoint can be used to filter incoming multicast from specific senders. Such
-a Preconnection will only support calling Listen(), not Initiate(). Calling Listen()
-will cause the Transport Services system to register for receiving multicast, such
-as issuing an IGMP join {{?RFC3376}} or using MLD for IPV6 {{?RFC4604}}. Any Connections
-that are accepted from this Listener are receive-only.
+To use multicast, a Preconnection is first created with the Remote Endpoint
+specifying the multicast group and destination port number.
 
-Similarly, specifying a multicast group address on the Remote Endpoint will indicate that the
-resulting connection will be used to send multicast messages, and that the Preconnection will
-support Initiate() but not Listen(). Any Connections created this way are send-only.
+```
+RemoteSpecifier.WithIPv4MulticastGroup(233.252.0.0)
+RemoteSpecifier.WithIPv6MulticastGroup(FF0X::DB8:0:0)
+LocalSpecifier.WithPort(5353)
+```
+
+Calling Initiate() on that Preconnection creates a Connection that can be
+used to send messages to the multicast group. The Connection object that is
+created will support Send() but not Receive(). Any Connections created this
+way are send-only, and do not join the multicast group. The resulting
+Connection will have a Local Endpoint indicating the local interface to
+which the connection is bound and a Remote Endpoint indicating the
+multicast group.
+
+Calling Listen() on a Preconnection with a multicast group specified on the Remote
+Endpoint will join the multicast group to receive messages. This Listener
+will create one Connection for each Remote Endpoint sending to the group,
+with the Local Endpoint set to the group address. The set of Connection
+objects created forms a Connection Group.
+
+Calling Rendezvous() on a Preconnection with an any-source multicast group
+address as the Remote Endpoint will join the multicast group, and also
+indicates that the resulting connection can be used to send messages to the
+multicast group. The Rendezvous() call will return both a Connection that
+can be used to send to the group, that acts the same as a connection
+returned by calling Initiate() with a multicast Remote Endpoint, and a
+Listener that acts as if Listen() had been called with a multicast Remote
+Endpoint.
+
+Calling Rendezvous() on a Preconnection with a source-specific multicast
+group address as the Remote Endpoint results in an EstablishmentError.
```

Comment:
SSM group address is not entirely knowable just from the address value, so I'm not sure this is well-founded.

For instance the administratively scoped space (239/8 in IP4 IIRC) I think can by design be either according to how the site wants to treat it, and in practice other addresses outside of 232/8 or ff3x::/96 can likewise generally be configured as ssm optionally in most routers.  And also ASM joins can be requested on the reserved ssm addresses.

So this condition also would be much better handled by prohibiting Rendevous() this way only according to an explicit SSM indicator as part of WithIPxMulticastGroup().  (Or similar, such as a WithSSM(true) or something, though I think it's probably better to make it 'WithIPxMulticastGroup(addr, SSM)', since I don't think an ASM/SSM indicator makes sense unless there is a MulticastGroup.